src/librustc_mir/build/matches/mod.rs

```diff
@@ -142,7 +142,7 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         // Step 2. Create the otherwise and prebinding blocks.
 
         // create binding start block for link them by false edges
-        let candidate_count = arms.iter().map(|c| c.patterns.len()).sum::<usize>();
+        let candidate_count = arms.iter().map(|c| c.top_pats_hack().len()).sum::<usize>();
```

Comment:
@dlrobertson You might want to link up with this (and the other diffs in this file) in https://github.com/rust-lang/rust/pull/63688 or leave it as a follow up.